samples/bluetooth/mesh_demo/prj_bbc_microbit.conf

```diff
@@ -5,7 +5,7 @@ CONFIG_SYSTEM_WORKQUEUE_STACK_SIZE=2048
 CONFIG_SYS_CLOCK_TICKS_PER_SEC=250
 CONFIG_GPIO=y
 CONFIG_FLASH=y
-CONFIG_SOC_FLASH_NRF5=y
+CONFIG_SOC_FLASH_NRF=y
```

Comment:
Is this needed with the auto selection based on CONFIG_FLASH?  Or maybe we set that up as a clean up patch later? (This would go for all of the sample *.conf files in this PR where CONFIG_FLASH is set)

Reply:
I have not tried without, but I think it's needed because `CONFIG_FLASH` does not autoselect the driver actually. Instead the driver depends on `CONFIG_FLASH` being enabled:

```
config SOC_FLASH_NRF
        bool "Nordic Semiconductor nRF5X flash driver"
        depends on FLASH && SOC_FAMILY_NRF
        default n
        select FLASH_HAS_PAGE_LAYOUT
        select FLASH_HAS_DRIVER_ENABLED
        help
          Enables Nordic Semiconductor nRF5X flash driver.
```

Reply:
I was looking at this code:
https://github.com/zephyrproject-rtos/zephyr/blob/master/arch/arm/soc/nordic_nrf/Kconfig.defconfig#L27

```
if FLASH

config SOC_FLASH_NRF
	def_bool y

endif # FLASH
```

Reply:
Oh yes, you are right, I had forgotten the SoC `.defconfig`. I can clean that up actually yes

Reply:
I just pushed an update removing the unnecessary extra definitions